plugins/SliceInfoPlugin/SliceInfo.py

```diff
@@ -112,19 +139,10 @@ def _onWriteStarted(self, output_device):
             submitted_data = urllib.parse.urlencode(submitted_data)
             binary_data = submitted_data.encode("utf-8")
 
-            # Submit data
-            kwoptions = {"data" : binary_data,
-                         "timeout" : 1
-                         }
-            if Platform.isOSX():
-                kwoptions["context"] = ssl._create_unverified_context()
-            try:
-                f = urllib.request.urlopen(self.info_url, **kwoptions)
-                Logger.log("i", "Sent anonymous slice info to %s", self.info_url)
-                f.close()
-            except Exception as e:
-                Logger.logException("e", "An exception occurred while trying to send slice information")
+            # Sending slice info non-blocking
+            reportJob = SliceInfoJob(self.info_url, binary_data)
+            reportJob.start()
         except:
             # We really can't afford to have a mistake here, as this would break the sending of g-code to a device
             # (Either saving or directly to a printer). The functionality of the slice data is not *that* important.
-            pass
+            Logger.logException("e", "Exception raised while sending slice info") # But we should be notified about these problems of course.
```

Comment:
I can hardly imaging ```pass``` is the command that takes time. It is the python equivalent of a "null operation".